### PR TITLE
Add Jazzy Docs

### PR DESCRIPTION
**Pull request by aamctustwo**
### Issue Link :link:

- #2224

### Goals :soccer:

- Add documentation to the repo rather than using (the now deprecated) CocoaDocs

### Implementation Details :construction:

- Adds `docs` folder with [Jazzy](https://github.com/realm/jazzy) generated documentation
- Adds Jazzy as a developer dependency (via Bundler)
- Update the readme to point to GitHub Pages (when it is configured in repo's settings) for locally generated Jazzy docs instead of CocoaDocs as it is now deprecated

### Testing Details :mag:

View documentation:

- Open `index.html` in the `docs` folder in a browser

Generate documentation:

- Run `bundle exec jazzy` in the root directory of the project

### Notes :warning:

- To use GitHub Pages for the Jazzy documentation, the repo's settings needs to be updated after merging into `master`. There's an option in the GitHub Pages configuration to adjust the `publishing source` to `master branch /docs folder`. See the [GitHub Pages documentation](https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/#publishing-your-github-pages-site-from-a-docs-folder-on-your-master-branch).

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+source "https://rubygems.org"
+
+gem "jazzy"
```
Comment (jshier) [inline Gemfile]: If we're adding a Ruby environment, can you add `cocoapods` to the `Gemfile`, and add `.ruby-version` (2.4.1) and `.ruby-gemset` (alamofire) files too? That way it will integrate with RVM / rbenv automatically.

File: .jazzy.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+author: Alamofire Software Foundation
+author_url: http://alamofire.org/
+github_url: https://github.com/Alamofire/Alamofire
+module: Alamofire
+output: docs
```
Comment (jshier) [inline .jazzy.yaml]: We probably want to refactor our Documentation layout with this. Perhaps we can put these docs in `Documentation/jazzy` and move the migration guides `Documentation/Migration Guides`. That will require some Xcode project changes for the guides, but there's no need to add the jazzy docs to Xcode at all.

File: Gemfile.lock [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      nanaimo (~> 0.2.3)
+
+PLATFORMS
+  ruby
+
+DEPENDENCIES
+  jazzy
+
+BUNDLED WITH
+   1.14.5
```
Comment (jshier) [inline Gemfile.lock]: Please use the latest `bundler`.

File: README.md [hunk 4]
```diff
@@ -3,7 +3,7 @@
 [![Build Status](https://travis-ci.org/Alamofire/Alamofire.svg?branch=master)](https://travis-ci.org/Alamofire/Alamofire)
 [![CocoaPods Compatible](https://img.shields.io/cocoapods/v/Alamofire.svg)](https://img.shields.io/cocoapods/v/Alamofire.svg)
 [![Carthage Compatible](https://img.shields.io/badge/Carthage-compatible-4BC51D.svg?style=flat)](https://github.com/Carthage/Carthage)
-[![Platform](https://img.shields.io/cocoapods/p/Alamofire.svg?style=flat)](http://cocoadocs.org/docsets/Alamofire)
+[![Platform](https://img.shields.io/cocoapods/p/Alamofire.svg?style=flat)](https://Alamofire.github.io/Alamofire)
```
Comment (jshier) [inline README.md]: It will be a lower case `alamofire.github.io`.

File: .jazzy.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+theme: fullwidth
+xcodebuild_arguments: [-workspace,'Alamofire.xcworkspace',-scheme,'Alamofire iOS']
```
Comment (cnoon) [inline .jazzy.yaml]: Spaces after the commas?

**Comment by cnoon** [comment]
So for my own clarification, would our release workflow change from the last commit being updating the version and finalizing the **CHANGELOG** to updating the Jazzy docs using `bundler`? I would assume we'd always want the docs to represent the latest officially released version...but maybe that's not a good assumption.

On a separate note, thanks for putting this together @aamctustwo! 🍻

File: README.md [hunk 6]
```diff
@@ -45,7 +45,7 @@ Alamofire is an HTTP networking library written in Swift.
 - [x] TLS Certificate and Public Key Pinning
 - [x] Network Reachability
 - [x] Comprehensive Unit and Integration Test Coverage
-- [x] [Complete Documentation](http://cocoadocs.org/docsets/Alamofire)
+- [x] [Complete Documentation](https://Alamofire.github.io/Alamofire)
```
Comment (cnoon) [inline README.md]: Lowercase as @jshier mentioned earlier.

File: .jazzy.yaml (hunk 2 above)
Comment (cnoon) [inline .jazzy.yaml]: After more investigation, it looks like it has to go in the `master` branch in a `docs` folder to work seamlessly with GitHub Pages. We can sort out where we put everything else in another PR. Might not really be that big of a deal to have a `docs` directly full of Jazzy docs and then a `Documentation` folder with everything else nested.

**Comment by aamctustwo** [comment]
@cnoon Yes, I'd recommend the final step to be updating the docs.

It might be worth raising a new issue to update the `CONTRIBUTING.md` file or add a new file in `docs`/`Documentation` to write out the release process as it begins to add more steps (as well as looking to possibly script some of the steps).

File: .jazzy.yaml (hunk 2 above)
Comment (jshier) [inline .jazzy.yaml]: Sounds good.

**Comment by cnoon** [comment]
We're live! http://alamofire.github.io/Alamofire/index.html
